File: day01/solution.jl
input = parse.(Int, readlines("input.txt"))

part1 = count(a < b for (a, b) in zip(input, input[2:end]))
println(part1)

part2 = count(a < b for (a, b) in zip(input, input[4:end]))
println(part2)

File: day01/solution.rs
use std::fs;

fn main() {
    let input = fs::read_to_string("input.txt").expect("Failed to read input.txt");
    let nums: Vec<usize> = input.lines().map(|s| s.parse::<usize>().unwrap()).collect();

    // Part 1
    let mut part1: usize = 0;
    for i in 0..nums.len() - 1 {
        if nums[i] < nums[i + 1] {
            part1 += 1;
        }
    }
    println!("{}", part1);

    // Part 2
    let mut part2: usize = 0;
    for i in 0..nums.len() - 3 {
        if nums[i] < nums[i + 3] {
            part2 += 1;
        }
    }
    println!("{}", part2);
}
